src/App.tsx

```diff
@@ -1,24 +1,373 @@
+
+    if (title.length < 1) {
+      deleteTodo(id);
+    } else {
+      updateTodo({
+        id, title, completed, userId,
+      });
+    }
+  }
+
+  function redactTodoHandler(todoId: number, todoTitle: string) {
+    setTodoNewTitle(todoTitle); // Спочатку встановлюємо todoNewTitle
+    setRedactedTodo(todoId); // Потім встановлюємо redactedTodo
+  }
+
+  // Пізніше, використовуючи useEffect, ви фокусуєте на введенні, коли redactedTodo змінюється
+  useEffect(() => {
+    if (redactedTodo !== null && todoInputRef.current) {
+      todoInputRef.current.focus();
+    }
+  }, [redactedTodo]);
+
+  useEffect(loadTodos, []);
+
+  function onDeleteCompleted() {
+    todos.filter(todo => todo.completed === true).map(todo => deleteTodo(todo.id));
+  }
+
+  function toggleAll() {
+    if (todos.some(todo => todo.completed === false)) {
+      todos.map(todo => updateTodo({
+        id: todo.id, title: todo.title, completed: true, userId: todo.userId,
+      }));
+    } else {
+      todos.map(todo => updateTodo({
+        id: todo.id, title: todo.title, completed: false, userId: todo.userId,
+      }));
+    }
+  }
+
+  const handleAddTodo = (event: FormEvent) => {
+    event.preventDefault();
+    setTempTodo({
+      title: newTodoTitle, completed: false, userId: USER_ID, id: 0,
+    });
+    if (!newTodoTitle) {
+      setErrorMsg('Title should not be empty');
+    } else {
+      addTodo({
+        title: newTodoTitle, completed: false, userId: USER_ID, id: 0,
+      });
+      setNewTodoTitle('');
+    }
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length > 0 && (
+            <button
+              type="button"
+              className={classNames('todoapp__toggle-all', { active: todos.every(todo => todo.completed === true) })}
+              data-cy="ToggleAllButton"
+              onClick={() => toggleAll()}
+            />
+          )}
+          <form onSubmit={handleAddTodo}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={newTodoTitle}
+              disabled={disInput}
+              ref={inputRef}
+              onChange={(event) => setNewTodoTitle(event.target.value)}
+            />
+          </form>
+        </header>
+        <section className="todoapp__main" data-cy="TodoList">
```

Comment:
try to split ypur code into components, you have `header`, `footer` , `TodoList` etc